dev/src/reference.ts

```diff
@@ -1035,6 +1047,33 @@ export class QueryOptions<T> {
     );
   }
 
+  /**
+   * Returns query options for a query that only fetches document names for all
+   * descendants under a specified reference.
+   *
+   * @private
+   */
+  static forDocumentNamesOnly<T>(
+    parent: ResourcePath,
+    id: string,
+    isDocumentPath: boolean
+  ): QueryOptions<T> {
+    let options = new QueryOptions<T>(
+      parent,
+      id,
+      defaultConverter(),
+      /*allDescendants=*/ true,
+      /*fieldFilters=*/ [],
+      /*fieldOrders=*/ []
+    );
+
+    options = options.with({
```

Comment:
Maybe use `select()` here (and maybe `FieldPath.documentId()`)

Reply:
Done. Also used `where()` for the MIN_ID logic to standardize in a single place.

Reply:
I don't want to add the extra setting to QueryOptions just to remove the `collectionId` field, but I can't figure out a clean way to. Any ideas?

Reply:
Can you add a `parent` to `forCollectionGroupQuery` and make the `collectionId` EMPTY (which removes `omitCollectionId` entirely)? 

FWIW, the query options object this method returns doesn't actually do the filtering of document names. If you keep this method, it should get a different name.

Reply:
Renamed to `forKindlessAllDescendants()`. 

We need the `collectionId` to not be empty since `Query.id` would then incorrectly return an empty string.